### DISCUSSION TITLE
Catching and reporting "not handled" navigation paths

### DISCUSSION THREAD

**Discussion by daveslutzkin** [react-navigation/react-navigation#10524] [Q&A] [answered]
In a production app, we always want to report any unexpected situations or other errors. So we push any errors we detect to an error monitoring/reporting service and triage them periodically.

React Navigation sometimes gives errors like this:
```
The action 'PUSH' with payload {"name":"something"} was not handled by any navigator.

Do you have a screen named 'something'?

If you're trying to navigate to a screen in a nested navigator, see https://reactnavigation.org/docs/nesting-navigators#navigating-to-a-screen-in-a-nested-navigator.

This is a development-only warning and won't be shown in production.
```
This obviously happens because a developer makes a mistake. That's fine as long as we find out about it.

However, there doesn't seem to be any way for us to detect and report on this situation using React Navigation.

Am I missing something obvious? Surely this is pretty important for a production app?

**Comment by raajnadar** [comment]
Read the last line in the warning it is dev only.

**Reply by daveslutzkin** [reply]
@raajnadar That's not helpful. I want the warning logged because then I find out that a user clicked on a link which didn't take them anywhere. That's a bad thing in most apps.

**Reply by raajnadar** [reply]
Warnings are always dev only, they are of no use on production. What is the use case? If you add a link you know that where it is pointing atleast in mobile apps.

**Reply by daveslutzkin** [reply]
Of course warnings are useful in production! That's why error logging tools exist.

A use case recently from our app was that a developer (OK, me) removed a seemingly unused screen without noticing that a link still existed to it. It took a couple of days to discover that issue because the 404 wasn't logged anywhere.

**Reply by raajnadar** [reply]
The warning thrown is not fatal (yellow box), errors are fatal and they cause app crashes (red box). You can try using strong type checking to prevent this issue.

**Reply by daveslutzkin** [reply]
@raajnadar I understand the difference between warnings and errors. My point is that it's bad for the users of an app if they tap on a link and it doesn't go anywhere.

**Comment by satya164** [comment, marked as answer]
`NavigationContainer` accepts a `onUnhandledAction` prop. However if you use TypeScript then it'll prevent such an issue.

**Reply by daveslutzkin** [reply]
It accepts `onUnhandledAction`? That's great! I can't see that anywhere in the docs.

**Reply by daveslutzkin** [reply]
@satya164 Thanks, this works perfectly, exactly the right solution. I'll try to find time to PR it into the docs.

**Comment by devcosta** [comment]
Hi, guys i have an issue. onUnhandledAction fire when the current router not found but, if you navigate on a nested navigation ​

ex:
`navigation.navigate('auth', { screen: 'nonExistentRouter' });`

onUnhandledAction dont fire because auth stack really exits but the child not

Can i have some solution that works in both cases?
